### PR TITLE
Command line interface modification

### PR DESCRIPTION
**Pull request by ryan-cahill**

### CODE REVIEW

File: src/commands/deploy.ts [hunk 1]
```diff
@@ -179,14 +185,34 @@ export default class Deploy extends DeployCommand {
...
+          [component_name]: tag,
+        },
+      });
+
+      dependency_manager = await LocalDependencyManager.create(this.app.api);
+      dependency_manager.environment = env_config;
+
+      const extra_interfaces = this.getExtraInterfaces(flags.interface);
+      this.updateEnvironmentInterfaces(env_config, extra_interfaces, component_name);
+      validate_params = false;
```
[the remark this answers is not in the document]
Comment (ryan-cahill) [reply src/commands/deploy.ts:203]: We're dynamically building the config when someone is deploying a single component, so there are no environment variables to override yet. If we try to validate one on component deployments, we're going to fail every time

[the remark this answers is not in the document]
Comment (tjhiggins) [reply src/commands/deploy.ts:203]: We need to set the parameters on the component in that scenario. 
```bash
architect deploy examples/react-app:latest -p world_text=override
```
will translate to:
```yaml
components:
  examples/react-app:
    extends: latest
    parameters:
      world_text: override
```

**Comment by dan-barrett** [comment]
@tjhiggins / @ryan-cahill 

Could we add a `-p` alias for [Platforms CLI commands](https://github.com/architect-team/architect-cli/blob/0c1248a4615026ce964830e71bd949cd4649bf13/src/common/utils/platform.ts#L15) like `-a` for [Accounts](https://github.com/architect-team/architect-cli/blob/0c1248a4615026ce964830e71bd949cd4649bf13/src/common/utils/account.ts#L15)

Or does that get too confusing with --parameter alias? We shouldn't be using them together correct? Really I just want -p in environment create (which is the only place we use PlatformUtils at the moment):

```
architect env:create -a examples -p k8s-local my-env
```

**Comment by ryan-cahill** [comment]
@dan-barrett we took the char alias from `--platform` and gave it to `--parameter` because there can be more than one of them and it would be slightly more convenient for parameters than platforms. Now that we've removed some of the magic of creating environments/specifying the platform on the deploy command, I don't see anywhere we'd find a collision. Personally I'd like to add that back to the `env:create` command

**Pull request merged**